File: src/main/java/net/tassia/util/Encoder.java
package net.tassia.util;

/**
 * An encoder is used to encode raw binary data.
 *
 * @since Commons 1.0
 * @author Tassilo
 */
public abstract class Encoder {

	/**
	 * Encodes the given data.
	 *
	 * @param data the data
	 * @return the encoded data
	 */
	public abstract String encode(byte[] data);

}

File: kommons-encoding/src/commonMain/kotlin/net.tassia.encoding/Encoder.kt
package net.tassia.encoding

/**
 * An encoder is used to encode raw binary data.
 *
 * @since Kommons 1.0
 * @author Tassilo
 */
interface Encoder {

	/**
	 * Encodes the given data.
	 *
	 * @param data the data
	 * @return the encoded data
	 */
	fun encode(data: ByteArray): String

}
